Fitness/Controllers/HomeViewController+Extensions.swift

```diff
@@ -0,0 +1,224 @@
+
+        switch sections[indexPath.section] {
+        case .checkIns:
+            let checkInListItemCellHeight = 53
+            let checkInListHeight = checkInListItemCellHeight * habits.count
+            let bottomSectionInset = 32
+            return CGSize(width: width, height: CGFloat(checkInListHeight + bottomSectionInset))
+        case .allGyms:
+            return CGSize(width: width, height: 123)
+        case .pros:
+            return CGSize(width: width, height: 145)
+        case .todaysClasses:
+            return CGSize(width: width, height: 227)
+        }
+    }
+
+    func collectionView(_ collectionView: UICollectionView, didHighlightItemAt indexPath: IndexPath) {
+        guard let cell = collectionView.cellForItem(at: indexPath) else { return }
+        cell.zoomIn()
+    }
+
+    func collectionView(_ collectionView: UICollectionView, didUnhighlightItemAt indexPath: IndexPath) {
+        guard let cell = collectionView.cellForItem(at: indexPath) else { return }
+        cell.zoomOut()
+    }
+
+}
+
+extension HomeViewController: GymsListCellDelegate {
+
+    func allGymsCellShouldOpenGym(_ gym: Gym) {
+        let gymDetailViewController = GymDetailViewController()
+        gymDetailViewController.gym = gym
+        navigationController?.pushViewController(gymDetailViewController, animated: true)
+    }
+
+}
+
+extension HomeViewController: TodaysClassesListCellDelegate {
+
+    func todaysClassesCellShouldOpenGymClassInstance(_ gymClassInstance: GymClassInstance) {
+        let classDetailViewController = ClassDetailViewController(gymClassInstance: gymClassInstance)
+        navigationController?.hidesBottomBarWhenPushed = true
+        navigationController?.pushViewController(classDetailViewController, animated: true)
+    }
+
+}
+
+extension HomeViewController: ProsListCellDelegate {
+
+    func prosListCellShouldOpenProBio(_ proBio: ProBio) {
+        let proDetailViewController = ProBioPageViewController()
+        proDetailViewController.pro = proBio
+        navigationController?.hidesBottomBarWhenPushed = true
+        navigationController?.pushViewController(proDetailViewController, animated: true)
+    }
+
+}
+
+extension HomeViewController: ChooseGymsDelegate {
+
+    func updateFavorites(favorites: [String]) {
+        favoriteGyms = favorites.compactMap { favorite in
+            self.gyms.first { $0.name == favorite }
+        }
+    }
+
+    func pushHabitOnboarding() {
+        let habitViewController = HabitTrackingController(type: .cardio)
+        self.navigationController?.pushViewController(habitViewController, animated: true)
+        return
+    }
+
+    func pushGymOnboarding() {
+        navigationController?.tabBarController?.tabBar.isHidden = true
+        let onboardingGymsViewController = OnboardingGymsViewController()
+        onboardingGymsViewController.delegate = self
+        navigationController?.pushViewController(onboardingGymsViewController, animated: true)
+        return
```

Comment:
does `return` at the end of this function and the above have a purpose?

Reply:
good catch